api/server/router/container/backend.go

```diff
@@ -43,7 +43,7 @@ type stateBackend interface {
 	ContainerStop(name string, seconds int) error
 	ContainerUnpause(name string) error
 	ContainerUpdate(name string, hostConfig *container.HostConfig) ([]string, error)
-	ContainerWait(name string, timeout time.Duration) (int, error)
+	ContainerWait(name string, timeout time.Duration, waitForRemoval bool) (int, error)
```

Comment:
The `waitForRemoval` argument for ContainerWait is pretty gross.
I'm curious why we can't wait for it to stop as normal and watch events for removal?

Honestly, I'm pretty -1 on this implementation for `ContainerWait`